pkg/client/client_oauth.go

```diff
@@ -214,28 +214,28 @@ func (c *OAuthClient) renewAccessToken() error {
 	return nil
 }
 
-// ParseCredentials reads credentials into a struct used. Performs validations.
-func ParseCredentials(data []byte) (*Credentials, error) {
-	var credentials Credentials
+// ParseOAuthCredentials reads credentials into an OAuthCredentials struct. Performs validations.
+func ParseOAuthCredentials(data []byte) (Credentials, error) {
```

Comment:
Nitpick: I'd expect a `ParseOAuthCredentials` to return a `OAuthCredentials` concrete struct directly rather than the interface. currently a caller calling ParseOAuthCredentials would have to do a typecheck for `OAuthCredentials` on the return item even if the method is called `ParseOAuthCredentials`.

Reply:
You're right, it defeats the purpose. I've updated it to return the correct struct.